src/sources/csv.rs

```diff
@@ -0,0 +1,261 @@
+//! CSV source for syncing with Famedly's Zitadel.
+
+use std::{fs, path::PathBuf};
+
+use anyhow::{Context, Result};
+use async_trait::async_trait;
+use csv::Reader;
+use serde::Deserialize;
+
+use super::Source;
+use crate::{user::User, zitadel::SourceDiff};
+
+/// CSV Source
+pub struct CsvSource {
+	/// CSV Source configuration
+	csv_config: CsvSourceConfig,
+}
+
+#[async_trait]
+impl Source for CsvSource {
+	fn get_name(&self) -> &'static str {
+		"CSV"
+	}
+
+	async fn get_diff(&self) -> Result<SourceDiff> {
+		let new_users = self.read_csv()?;
+		return Ok(SourceDiff { new_users, changed_users: vec![], deleted_user_ids: vec![] });
+	}
+}
+
+impl CsvSource {
+	/// Create a new CSV source
+	pub fn new(csv_config: CsvSourceConfig) -> Self {
+		Self { csv_config }
+	}
+
+	/// Get list of users from CSV file
+	fn read_csv(&self) -> Result<Vec<User>> {
+		let file_path = &self.csv_config.file_path;
+		let file = fs::File::open(&self.csv_config.file_path)
+			.context(format!("Failed to open CSV file {}", file_path.to_string_lossy()))?;
+		let mut reader = Reader::from_reader(file);
+		let users: Result<Vec<CsvData>, _> = reader.deserialize().collect();
+		let users: Result<Vec<User>, _> = users
+			.context("Failed to deserialize CSV data")
+			.map(|users| users.into_iter().map(CsvData::to_user).collect());
+		users
+	}
+}
+
+/// Configuration to get a list of users from a CSV file
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+pub struct CsvSourceConfig {
+	/// The path to the CSV file
+	pub file_path: PathBuf,
+}
+
+/// CSV data structure
+#[derive(Debug, Deserialize)]
+struct CsvData {
+	/// The user's email address
+	email: String,
+	/// The user's first name
+	first_name: String,
+	/// The user's last name
+	last_name: String,
+	/// The user's preferred username
+	preferred_username: String,
```

Comment:
We basically only record this to add it to metadata to remember it since it's available on ldap, it's never used in practice. Do we actually need to define it for the CSV source?

Reply:
May need to ask product for specifics here.

Reply:
Yes, looks like it's kept for linking LDAP user to Zitadel user and it's quite entangled in the old code, specifically the User struct and everywhere it's used. I will see to what would be the best way to separate it. Probably making it an Option so that LDAP still can use it, but other sources don't have to.

Reply:
Done. I made the preferred_username being the email address by default if not supplied by LDAP otherwise.

Reply:
*Technically* I think it'd be best if we didn't set this field at all when the CSV source is used. I'm ok with either implementation though, given we don't actually know why we're setting this field at the moment.